exes/ethercat/inc/tfc/ec/devices/beckhoff/EL1xxx_impl.hpp

```diff
@@ -32,24 +32,17 @@ template <typename manager_client_type,
           stx::basic_fixed_string name,
           template <typename, typename>
           typename signal_t>
-void el1xxx<manager_client_type, size, entries, pc, name, signal_t>::process_data(std::span<std::byte> input,
-                                                                                  std::span<std::byte>) noexcept {
-  constexpr size_t minimum_byte_count = (size / 9) + 1;
-  if (input.size() != minimum_byte_count) {
-    logger_.warn("EL1XXX Size mismatch between process data and expected expected: {}, actual: {}", minimum_byte_count,
-                 input.size());
-    return;
-  }
-
+void el1xxx<manager_client_type, size, entries, pc, name, signal_t>::pdo_cycle(input_pdo const& input,
+                                                                               std::span<std::uint8_t>) noexcept {
   // Loop bytes
-  for (size_t byte = 0; byte < minimum_byte_count; byte++) {
+  for (auto byte : input) {
```

Comment:
This breaks the line 39 and line 41,
it is using the byte counter to jump over whole bytes.

Reply:
We really need to write tests for these classes so modifications 
become easier.

Reply:
It looks like there are some tests and they seem to be failing, this could be the reason why. @jbbjarnason

Reply:
nicely caught, I have changed this back